ast/Trees.h

```diff
@@ -194,15 +194,23 @@ class ExpressionPtr {
     }
 
     explicit operator bool() const noexcept {
-        return get() != nullptr;
+        return ptr != 0;
```

Comment:
We reset to nullptr when we move:

https://github.com/sorbet/sorbet/blob/615a299e1e6277bf3941ed0dffcdf7a1a81e41f5/ast/Trees.h#L147-L158
https://github.com/sorbet/sorbet/blob/615a299e1e6277bf3941ed0dffcdf7a1a81e41f5/ast/Trees.h#L105-L110

but that does not preclude the possibility of getting constructed with a nullptr and a bogus tag.  I guess we could add an `ENFORCE` on the tagging constructor or on `get()` to ensure that we're not winding up with weird things.  (Though the only place we call the tagging constructor is in `make_tree` and we cannot pass in a nullptr there because `new` would have thrown otherwise.)

(Could we have once upon a time wound up with tagged nullptr before we had `EmptyTree`?)